ctapipe/image/cleaning.py

```diff
@@ -67,16 +73,123 @@ def tailcuts_clean(geom, image, picture_thresh=7, boundary_thresh=5,
     # matrix (2d), we find all pixels that are above the boundary threshold
     # AND have any neighbor that is in the picture
     pixels_above_boundary = image >= boundary_thresh
-    pixels_with_picture_neighbors = geom.neighbor_matrix_sparse.dot(
-        pixels_in_picture)
+    pixels_with_picture_neighbors = geom.neighbor_matrix_sparse.dot(pixels_in_picture)
     if keep_isolated_pixels:
-        return (pixels_above_boundary
-                & pixels_with_picture_neighbors) | pixels_in_picture
+        return (
+            pixels_above_boundary & pixels_with_picture_neighbors
+        ) | pixels_in_picture
     else:
         pixels_with_boundary_neighbors = geom.neighbor_matrix_sparse.dot(
-            pixels_above_boundary)
-        return ((pixels_above_boundary & pixels_with_picture_neighbors) |
-                (pixels_in_picture & pixels_with_boundary_neighbors))
+            pixels_above_boundary
+        )
+        return (pixels_above_boundary & pixels_with_picture_neighbors) | (
+            pixels_in_picture & pixels_with_boundary_neighbors
+        )
+
+
+def mars_image_cleaning(
+    geom,
+    image,
+    picture_thresh=7,
+    boundary_thresh=[5, 5],
+    keep_isolated_pixels=False,
+    min_number_picture_neighbors=0,
+):
+    """
+    Clean an image by selection pixels that pass a three-threshold tail-cuts
+    procedure.
+    All thresholds are defined with respect to the pedestal
+    dispersion. All pixels that have a signal higher than the core (picture)
+    threshold will be retained, along with all those above the boundary
+    threshold that are neighbors of a core pixel AND all those above
+    the boundary threshold that are neighbors of a neighbor of a core pixel.
```

Comment:
In case any reference like a paper or proceeding exists, which you might have used in order to implement this, it might be nice to include the reference here for interested colleagues. 

In addition, in case you would rather refer to the original MAGIC implemenation (which is closed source, as far as I know) I believe it might still be helpful to include a reference here. Just in case someone is interested in comparing to the original C++ code.

Reply:
Dear Dominik,

for this specific case this is simply orally-distributed information.
I don't know wich reference it should be used exactly, I will ask Abelardo.

Cheers,
Michele

Reply:
thats nice of you, thanks.

Reply:
An update on this: Abelardo told that he doesn't have a reference at the moment, since what he did is the result of a inspiration from EventDisplay (in fact, MAGIC doesn't use double pass cleaning). I will contact the responsible from EventDisplay.